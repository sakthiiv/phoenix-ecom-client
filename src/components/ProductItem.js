import React, { Component } from "react";
import { Redirect, Link } from "react-router-dom";
import {
  Card,
  CardText,
  CardBody,
  Row,
  Col
} from "reactstrap";

class ProductItem extends React.Component {
  constructor() {
    super();
    // this.state = { categories: [] };
  }

  componentDidMount() {
    
  }
  redirectToPList = (e) => {
    e.preventDefault();
    this.props.redirectToPList();
  }
  render() {
    const {productItem} = this.props; 
    return (
        <Link to={{ pathname: '/product-detail', state: { product: productItem} }}>
            <div className="list-container prod-list" key={productItem.id}>
                <Row>
                    <Col sm="12">
                        <Card>
                            <picture class="img-product">
                                <img src={"http://10.132.20.169:8080/api/v1/image/" + productItem.id + ".jpg"}/>
                            </picture>
                            <div class="prod-body-container">
                                <CardBody>
                                    <div className="cart-title-container">
                                        <div className="cart-title-left">
                                            <h3>{productItem.name}</h3>
                                            <p>{productItem.description}</p>
                                            <p>{productItem.price}$</p>
                                        </div>
                                    </div>
                                </CardBody>
                            </div>
                        </Card>
                    </Col>
                </Row>
            </div>
        </Link>
    );
  }
}

export default ProductItem;
