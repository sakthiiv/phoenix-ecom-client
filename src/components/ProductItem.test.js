import React from "react";
import { shallow } from "enzyme";
import ProductItem from "./ProductItem";

const product = {
  "categoryId": "5bc9a87d76e3833f73511a17",
  "description": "asdasd",
  "id": "5bc9d01964151f13906681b7",
  "name": "Sony bravia",
  "price": 20000,
  "subCategoryId": "2"
  };
describe("ProductItem", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ProductItem productItem={product}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
