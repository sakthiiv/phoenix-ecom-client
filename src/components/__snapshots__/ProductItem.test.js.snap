// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders correctly 1`] = `
<Link
  replace={false}
  to={
    Object {
      "pathname": "/product-detail",
      "state": Object {
        "product": Object {
          "categoryId": "5bc9a87d76e3833f73511a17",
          "description": "asdasd",
          "id": "5bc9d01964151f13906681b7",
          "name": "Sony bravia",
          "price": 20000,
          "subCategoryId": "2",
        },
      },
    }
  }
>
  <div
    className="list-container prod-list"
    key="5bc9d01964151f13906681b7"
  >
    <Row
      tag="div"
    >
      <Col
        sm="12"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Card
          tag="div"
        >
          <picture
            class="img-product"
          >
            <img
              src="http://10.132.20.169:8080/api/v1/image/5bc9d01964151f13906681b7.jpg"
            />
          </picture>
          <div
            class="prod-body-container"
          >
            <CardBody
              tag="div"
            >
              <div
                className="cart-title-container"
              >
                <div
                  className="cart-title-left"
                >
                  <h3>
                    Sony bravia
                  </h3>
                  <p>
                    asdasd
                  </p>
                  <p>
                    20000
                    $
                  </p>
                </div>
              </div>
            </CardBody>
          </div>
        </Card>
      </Col>
    </Row>
  </div>
</Link>
`;
